rust/private/clippy.bzl

```diff
@@ -94,15 +94,22 @@ def _clippy_aspect_impl(target, ctx):
         maker_path = clippy_marker.path,
     )
 
-    # Deny the default-on clippy warning levels.
-    #
-    # If these are left as warnings, then Bazel will consider the execution
-    # result of the aspect to be "success", and Clippy won't be re-triggered
-    # unless the source file is modified.
-    args.add("-Dclippy::style")
-    args.add("-Dclippy::correctness")
-    args.add("-Dclippy::complexity")
-    args.add("-Dclippy::perf")
+    # Turn any warnings from clippy or rustc into an error, as otherwise
+    # Bazel will consider the execution result of the aspect to be "success",
+    # and Clippy won't be re-triggered unless the source file is modified.
+    if "__bindgen" in ctx.rule.attr.tags:
+        # bindgen-generated content is likely to trigger warnings, so
+        # only fail on clippy warnings
+        args.add("-Dclippy::style")
+        args.add("-Dclippy::correctness")
+        args.add("-Dclippy::complexity")
+        args.add("-Dclippy::perf")
+    else:
+        # fail on any warning
+        args.add("-Dwarnings")
```

Comment:
Does this actually trigger clippy lints?

Reading https://github.com/rust-lang/rust-clippy#clippy , it seems like "-Dclippy::all" would need to be passed in *addition* to "-Dwarnings", no?

Reply:
Clippy lints are a subclass of warnings:

```
bazel build --aspects=@io_bazel_rules_rust//rust:rust.bzl%rust_clippy_aspect --output_groups=clippy_checks ...
DEBUG: /private/var/tmp/_bazel_dae/03e4b12a9fa9e97f4f0ecc617a9bcea7/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: buildkite_config not using checked in configs; Bazel version 3.7.0 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"]}
INFO: Analyzed 9 targets (0 packages loaded, 0 targets configured).
INFO: Found 9 targets...
INFO: From Clippy test/clippy/bad_binary_clippy.ok:
error: literal with an empty format string
 --> test/clippy/bad_src/main.rs:3:24
  |
3 |         println!("{}", "Hello World");
  |                        ^^^^^^^^^^^^^
  |
  = note: `-D clippy::print-literal` implied by `-D clippy::style`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#print_literal
[...]
```